pkg/internal/affinityassistant/transformer_test.go

```diff
@@ -88,3 +88,143 @@ func TestNewTransformer(t *testing.T) {
 		})
 	}
 }
+
+func TestNewTransformerWithNodeAffinity(t *testing.T) {
+
+	nodeAffinity := &corev1.NodeAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+			NodeSelectorTerms: []corev1.NodeSelectorTerm{{
+				MatchFields: []corev1.NodeSelectorRequirement{{
+					Key:      "kubernetes.io/hostname",
+					Operator: corev1.NodeSelectorOpNotIn,
+					Values:   []string{"192.0.0.1"},
+				}},
+			}},
+		},
+	}
+
+	podAffinityTerm := &corev1.PodAffinityTerm{
+		LabelSelector: &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"from": "podtemplate",
+			},
+		},
+		TopologyKey: "kubernetes.io/hostname",
+	}
+
+	for _, tc := range []struct {
+		description string
+		annotations map[string]string
+		pod         *corev1.Pod
+		expected    *corev1.Affinity
+	}{{
+		description: "no affinity annotation and pod contains nodeAffinity",
+		annotations: map[string]string{
+			"foo": "bar",
+		},
+		pod: &corev1.Pod{
+			Spec: corev1.PodSpec{
+				Affinity: &corev1.Affinity{
+					NodeAffinity: nodeAffinity,
+				},
+			},
+		},
+		expected: &corev1.Affinity{
+			NodeAffinity: nodeAffinity,
+		},
+	}, {
+		description: "affinity annotation and pod contains nodeAffinity",
+		annotations: map[string]string{
+			"foo": "bar",
+			workspace.AnnotationAffinityAssistantName: "baz",
+		},
+		pod: &corev1.Pod{
+			Spec: corev1.PodSpec{
+				Affinity: &corev1.Affinity{
+					NodeAffinity: nodeAffinity,
+				},
+			},
+		},
+
+		expected: &corev1.Affinity{
+			NodeAffinity: nodeAffinity,
+			PodAffinity: &corev1.PodAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{{
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							workspace.LabelInstance:  "baz",
+							workspace.LabelComponent: workspace.ComponentNameAffinityAssistant,
+						},
+					},
+					TopologyKey: "kubernetes.io/hostname",
+				}},
+			},
+		},
+	}, {
+		description: "affinity annotation with a different name and pod contains podAffinity",
```

Comment:
it might also be useful to have a test case for pod affinity with preferredduringschedulingignoredduringexecution